sql/descriptor.go

```diff
@@ -72,6 +72,32 @@ func (p *planner) writeDescriptor(key proto.Key, descriptor descriptorProto, ifN
 	})
 }
 
+// deleteDescriptor delete the descriptor at `key`.
+func (p *planner) deleteDescriptor(key proto.Key) error {
+	// TODO(pmattis): The error currently returned below is likely going to be
+	// difficult to interpret.
+	// TODO(XisiHuang): Need to handle if-exists here as well.
+	return p.db.Txn(func(txn *client.Txn) error {
```

Comment:
We have an open issue (#1889) to plumb transaction support throughout the SQL statement execution. I think I'd leave off the `p.db.Txn` stuff for now.